config/config.go

```diff
@@ -101,3 +119,13 @@ func WithOpenTelemetryConfiguration(cfg OpenTelemetryConfiguration) Configuratio
 
 // TODO: create SDK from the model:
 // - https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4371
+
+func newResource(res *Resource) (*resource.Resource, error) {
+	if res == nil {
+		return resource.Default(), nil
+	}
+	return resource.Merge(resource.Default(),
+		resource.NewWithAttributes(semconv.SchemaURL,
+			semconv.ServiceName(*res.Attributes.ServiceName),
+		))
```

Comment:
How about moving to `resource.go` and adding tests in `resource_test.go`?

Can we do it in a separate PR?

Reply:
Sure, I'll move this next before the meter provider PR

Reply:
https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4821